pkg/webhooks/imagecontentpolicies/imagecontentpolicies.go

```diff
@@ -0,0 +1,188 @@
+package imagecontentpolicies
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName = "imagecontentpolicies-validation"
+	WebhookDoc  = "Managed OpenShift customers may not create ImageContentPolicy or ImageContentSourcePolicy resources that configure mirrors for quay.io, registry.redhat.com, nor registry.access.redhat.com."
+	// unauthorizedRepositoryMirrors is a regex that is used to reject certain specified repository mirrors.
+	// Generally all contained regexes follow a similar pattern, i.e. rejecting quay.io or quay.io/.*
+	unauthorizedRepositoryMirrors = `(^quay\.io(/.*)?$|^registry\.redhat\.io(/.*)?$|^registry\.access\.redhat\.com(/.*)?)`
+)
+
+type ImageContentPoliciesWebhook struct {
+	scheme *runtime.Scheme
+	log    logr.Logger
+}
+
+func NewWebhook() *ImageContentPoliciesWebhook {
+	return &ImageContentPoliciesWebhook{
+		scheme: runtime.NewScheme(),
+		log:    logf.Log.WithName(WebhookName),
+	}
+}
+
+func (w *ImageContentPoliciesWebhook) Authorized(request admission.Request) admission.Response {
+	decoder, err := admission.NewDecoder(w.scheme)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	icp := configv1.ImageContentPolicy{}
+	icsp := operatorv1alpha1.ImageContentSourcePolicy{}
+	allowedResp := admission.Response{
+		AdmissionResponse: admissionv1.AdmissionResponse{
+			UID:     request.UID,
+			Allowed: true,
+		},
+	}
+
+	if len(request.Object.Raw) > 0 {
```

Comment:
What is the importance of when the length of the raw object is greater than 0? It looks like we're Doing Things ™️ when it is the case, but when it is not greater than 0, the request is allowed by virtue of lines [47-52](https://github.com/openshift/managed-cluster-validating-webhooks/pull/213/files#diff-9697b102dd57fb098f8ff05ef470c690297d5f5fe218980f456b7b4e7cc0a02dR48-R52)?

Reply:
The length should never be 0 for a "valid" request since it means there's no object in the request. It's more of a guard before we attempt to check `request.RequestKind.Kind` or importantly before we attempt to `decoder.Decode(request, &kind)`

Reply:
Should this logic check move to `Validate`?